app/stac_api/utils.py

```diff
@@ -50,6 +52,25 @@ def get_link(links, rel, raise_exception=False):
     return None
 
 
+def decode_path_if_base64(base64_path):
+    '''Returns a ascii text as in the input or it encodes the base64 input
+
+    Args:
+        base64_path: base64 or ascii
+            A string, maybe decoded in base64
+    Return:
+        A a utf-8 encoded string
+    '''
+    try:
+        # decode base64
+        base64_bytes = base64_path.encode('ascii')
+        message_bytes = base64.b64decode(base64_bytes)
+        path = message_bytes.decode('ascii')
+    except (TypeError, binascii.Error) as error:
+        path = base64_path
+    return path
+
+
```

Comment:
No need of this the encoding should be done outside the service and no need to decode.